src/nvml/hook.c

```
@@ -353,18 +353,28 @@ nvmlReturn_t _nvmlDeviceGetMemoryInfo(nvmlDevice_t device,void* memory,int versi
             return NVML_SUCCESS;
         }
     } else {
+        size_t clamped = (usage > limit) ? limit : usage;
+        if (usage > limit) {
+            LOG_WARN("NVML meminfo: usage %lu exceeds limit %lu, clamping", usage, limit);
```

Cppcheck (2.21.0)

[warning] If resource allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfResources)